src/drive/web/modules/navigation/duck/reducer.js

```diff
@@ -358,20 +358,17 @@ export const getFileById = ({ view }, id) => {
   return ensureFileHavePath({ view }, file)
 }
 
-const isRootFolder = folder => folder.id === ROOT_DIR_ID
-
 const ensureFileHavePath = ({ view }, file) => ({
   ...file,
   path: getFilePath({ view }, file)
 })
 
 export const getFilePath = ({ view }, file) => {
   const { displayedFolder } = view
-  return isDirectory(file)
+  const isRecentView = displayedFolder === null
```

Comment:
Continuing the discussion  started here https://github.com/cozy/cozy-drive/commit/cf88e57c149943b77d32870226cca69f7e4b8228#r31858308

The `displayedFolder ` being `null` is indeed not enough to mean we're in the recent view. I think the sharing view for example also has no `displayedFolder` set.

But generally speaking, could we do something like this?

```
return file.path || `${displayedFolder ? displayedFolder.path : ''}/${file.name}`.replace('//', '/')
```

Reply:
In case of the sharing view, we want the same behaviour (even though it's not done yet:  https://trello.com/c/7bV17E2g )

As I said above I thought about the solution you suggest but I'm afraid that the `file.path` condition may create regressions. If you think it's ok, I'm ok with this solution. Also, I still don't understand why in some cases we want the file name in the file path and in other cases we don't.

Reply:
Since we will do it for the sharing view, we can rename the proposed `isRecentView` by something more explicit ? 
```js 
const viewsThatNeedToDisplayPath = ['recent', 'sharings'];
if(viewsThatNeedToDisplayPath.includes(currentView)){
...
}
```

Also, I'm not a big fan of the `.replace('//', '/')` regex :(